rust/src/lib/nm/nm_dbus/connection/route.rs

```diff
@@ -26,18 +30,32 @@ pub struct NmIpRoute {
 impl TryFrom<DbusDictionary> for NmIpRoute {
     type Error = NmError;
     fn try_from(mut v: DbusDictionary) -> Result<Self, Self::Error> {
+        let dest = _from_map!(v, "dest", String::try_from)?;
+        let default_next_hop = dest.as_deref().and_then(|dest| {
+            match dest.parse::<std::net::IpAddr>() {
```

Comment:
Base on your code it seems you are assuming `dest` in NM is always 0.0.0.0 or ::. Is this correct?

Reply:
And `nm_dbus` should not have this modification code, it should be simple dbus wrapper. This polish/modification should be done at `nm` level.

Reply:
No. Internally, nmstate represents "empty next-hop" as `Some("0.0.0.0")` for IPv4 and `Some("::")` for IPv6. So what I do here is detecting whether it's IPv4 or 6 depending on `dest`, so I can know what the value for "empty next-hop" should be.

Nmstate is doing this because `None` means "ignore next-hop value for matching", not "empty next-hop".

Where can I do the modification at `nm` level? This is where the `NmIpRoute`'s are created, so what I am doing is to create them always with an unified criteria for "default" or "empty" values.

Reply:
I don't see how to do this at `nm` level. What it's failing is the comparison between `NmIpRoutes`. This is only defined at `nm_dbus` level and it's almost not used outside of `nm_dbus`. Only a few minor usages from `nm/query_apply` and `nm/settings`.

Reply:
Fixed all at `nm` level. I've left a small normalization in `nm_dbus`, but it's a condition that should never happen and I've added a warning message so we can notice it it starts happening at some point, so it will be easy to debug.